### PR TITLE
drain response body to make sure connection reuse

### PR DESCRIPTION
**Pull request by yingchen0706v**
* drain response body to make sure connection reuse
* Also removed a duplicate error checking.

### CODE REVIEW

File: vast/vastutil/unwrap.go [hunk 1]
```diff
@@ -63,13 +65,12 @@ func unwrap(ctx context.Context, v *vast.Vast, unwrappedList []*vast.Vast, ua, i
...
-			resp.Body.Close()
+			_, _ = io.Copy(ioutil.Discard, resp.Body)
+			_ = resp.Body.Close()
 		}
 	}()
 
-	if err != nil {
-		return nil, err
-	} else if err = xml.NewDecoder(resp.Body).Decode(&innerVast); err != nil {
+	if err = xml.NewDecoder(resp.Body).Decode(&innerVast); err != nil {
```
Comment (xiaxiao-vungle) [inline vast/vastutil/unwrap.go:73]: if change to this way would cause the original err was not returned?

**Review by xiaxiao-vungle — DISMISSED** [review]

**Comment by xiaxiao-vungle** [comment]
LGTM

File: vast/vastutil/unwrap.go (hunk 1 above)
Comment (yingchen0706v) [reply vast/vastutil/unwrap.go:73]: as it is duplicate with line 63

**Review by xiaxiao-vungle — APPROVED** [review]

**Comment by xiaxiao-vungle** [comment]
LGTM

**Pull request closed without merging**